### PR TITLE
Move bower_components libraries from source control to npm

### PR DESCRIPTION
**Pull request by grafixeyehero**
- [x] alameda 1.3.0

-  [x] requirejs 2.3.5

### CODE REVIEW

**Comment by dkanada** [comment]
I think we can avoid pagejs and query string for now since those two require a lot more work than the rest. We can focus on this one when some of your earlier pull requests get merged in, there's a bit of a backlog.

File: assets.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+//const CSS = [
+
+//];
```
Comment (dkanada) [inline assets.js]: I'd remove this commented out section.

File: assets.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+const JS = [
+    'requirejs/require.js',
+    'alameda/alameda.js',
+    'webcomponents.js-2/webcomponents-lite.min.js',
+    'document-register-element/build/document-register-element.js',
+    'native-promise-only/lib/npo.src.js',
+    'resize-observer-polyfill/dist/ResizeObserver.js',
+    'whatwg-fetch/fetch.js'
```
Comment (dkanada) [inline assets.js]: I think alameda and requirejs are the only two that need this special method. The rest can just use the bundle method that @thornbill started using.

File: webpack.common.js [hunk 3]
```diff
@@ -10,9 +14,18 @@ module.exports = {
...
-        }])
+        }]),
+        new CopyPlugin(
+            Assets.map(asset => {
+              return {
+                from: path.resolve(__dirname, `./node_modules/${asset}`),
+                to: path.resolve(__dirname, './dist/bower_components')
+              };
+            })
+          )
```
Comment (dkanada) [inline webpack.common.js]: Something is weird with the indentation here...

File: assets.js (hunk 1 above)
Comment (grafixeyehero) [reply assets.js]: https://github.com/jellyfin/jellyfin-web/pull/549/commits/651a53d64a2d44abaa7bfd9bf1bf58e5a7fe4122

File: webpack.common.js (hunk 3 above)
Comment (grafixeyehero) [reply webpack.common.js]: https://github.com/jellyfin/jellyfin-web/pull/549/commits/651a53d64a2d44abaa7bfd9bf1bf58e5a7fe4122

File: assets.js (hunk 2 above)
Comment (grafixeyehero) [reply assets.js]: https://github.com/jellyfin/jellyfin-web/pull/549/commits/651a53d64a2d44abaa7bfd9bf1bf58e5a7fe4122

File: webpack.common.js [hunk 4]
```diff
@@ -10,9 +14,18 @@ module.exports = {
...
         new CopyPlugin([{
             from: "**/*",
             to: "."
-        }])
+        }]),
+        new CopyPlugin(
+            Assets.map(asset => {
+                return {
+                    from: path.resolve(__dirname, `./node_modules/${asset}`),
+                    to: path.resolve(__dirname, './dist/bower_components')
```
Comment (thornbill) [inline webpack.common.js]: I would prefer this directory be named something else to differentiate between resources that are still in `bower_components`.

Comment (dkanada) [reply webpack.common.js]: How about `dist/libraries` for things we need copied directly?

Comment (grafixeyehero) [reply webpack.common.js]: https://github.com/jellyfin/jellyfin-web/pull/549/commits/af2b49ac303d36de2c107a0eeb98968679609cce

File: webpack.common.js [hunk 5]
```diff
@@ -1,6 +1,10 @@
 const path = require("path");
+const { CleanWebpackPlugin} = require("clean-webpack-plugin");
 const CopyPlugin = require("copy-webpack-plugin");
 
+// assets.js
+const Assets = require('./assets');
```
Comment (thornbill) [inline webpack.common.js]: Should we just inline the assets here? I can't think of an advantage to having them in a separate file.

Comment (grafixeyehero) [reply webpack.common.js]: @thornbill you say like this 
`const Assets = [
    'alameda/alameda.js',
    'requirejs/require.js'
];`

Comment (thornbill) [reply webpack.common.js]: Yeah I prefer that personally.

Comment (grafixeyehero) [reply webpack.common.js]: https://github.com/jellyfin/jellyfin-web/pull/549/commits/863eb067c22b45a1416800d3dc2667d896dbf12a

**Pull request merged**